### DISCUSSION TITLE
Remote storage for workers

### DISCUSSION THREAD

**Discussion by nicolasiltis** [PrefectHQ/prefect#10277] [Ideas] [answered]
Hi Prefect Team !

I understand with this [PR](https://github.com/PrefectHQ/prefect/pull/8619) that for the first version of the worker it isn't possible to run a deployment on a worker with a storage block, it has to be a local storage. ([Related code](https://github.com/PrefectHQ/prefect/blob/main/src/prefect/workers/base.py#L771))

Will this change in the future ? If yes, do you have an ETA for this feature ? 

Thanks a lot,
Nicolas

**Comment by tekumara** [comment]
Likewise wondering this as the error suggests this is a temporary thing:

> ValueError: Flow run UUID('ad8cdf79-9d7b-41dc-a26d-09c5ae01306b') was created from deployment 'python' which is configured with a storage block. Workers **currently** only support local storage. Please use an agent to execute this flow run.

**Comment by billpalombi** [comment, marked as answer]
Yes, workers will support remote storage options in the future. While we don't have a firm timeline to shipping that functionality, my 95% CI is something like 5 -10 weeks.
